Makefile

```diff
@@ -5,18 +5,18 @@ default: build/universal-robots
 format:
 	ls src/*.*pp main.cpp | xargs clang-format-15 -i --style=file
 
-build: 
+build:
 	mkdir build
 
 test: build/universal-robots
 	./build/universal-robots-test
 
 build/universal-robots: build
-	cd build && \
-	cmake -G Ninja  .. && \
-	ninja all -j 4
+	cmake -S . -B build -G Ninja
+	cmake --build build --target all -- -j4
+	cmake --install build --prefix build/install/usr
```

Comment:
As above, CMake can act as the driver for the configure and build steps, and use `cmake --install` to place it in `build/install/usr`. This `prefix` is actually a bit of a lie, since the files are actually going to get staged into the appimage under `/usr/`. That's OK actually, since `/usr/lib` and friends are on the default library search path.

I'm somewhat interested in taking a closer look at this, because what I actually want is:

`cmake --install build --prefix /usr --destdir build/install`. In other words, put an installation designed for placement in `/usr` into `build/install`.